services/venice-controller/src/main/java/com/linkedin/venice/controller/stats/SparkServerStats.java

```diff
@@ -36,14 +36,11 @@ public class SparkServerStats extends AbstractVeniceStats {
   private final MetricEntityStateFourEnums<ControllerRoute, HttpResponseStatusEnum, HttpResponseStatusCodeCategory, VeniceResponseStatusCategory> successfulRequestLatencyHistogramMetric;
   private final MetricEntityStateFourEnums<ControllerRoute, HttpResponseStatusEnum, HttpResponseStatusCodeCategory, VeniceResponseStatusCategory> failedRequestLatencyHistogramMetric;
 
-  public SparkServerStats(MetricsRepository metricsRepository, String clusterName) {
-    super(metricsRepository, clusterName);
+  public SparkServerStats(MetricsRepository metricsRepository, String statsPrefix, String clusterName) {
+    super(metricsRepository, clusterName + "." + statsPrefix);
```

Comment:
should it be `clusterName.equals("NON_CLUSTER_SPECIFIC_STAT_CLUSTER_NAME") ? statsPrefix : clusterName + "." + statsPrefix`?

can you add tests for these?